### PR TITLE
Add obscureText field to TextInputConfiguration

### PR DESCRIPTION
**Pull request by cbracken**

### CODE REVIEW

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -46,19 +46,24 @@ class TextInputConfiguration {
...
+    this.obscureText: false,
     this.actionLabel,
-  }) : assert(inputType != null);
+  }) : assert(inputType != null),
+       assert(obscureText != null);
 
   /// The type of information for which to optimize the text input control.
   final TextInputType inputType;
 
+  final bool obscureText;
```
Comment (Hixie) [inline packages/flutter/lib/src/services/text_input.dart]: docs?

**Comment by Hixie** [comment]
test?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/services/text_input.dart (hunk 1 above)
Comment (cbracken) [inline packages/flutter/lib/src/services/text_input.dart]: Done.

**Comment by cbracken** [comment]
Added a test.

**Comment by cbracken** [comment]
Partial fix for #9411 and #6760

**Pull request merged**